inc/ix/drivers.h

```diff
@@ -34,3 +34,17 @@ extern struct rte_eth_txconf tx_conf;
 
 /* driver specific initialization functions */
 int ixgbe_init(struct ix_rte_eth_dev *dev, const char *driver_name);
+
+/* driver-independent eth_dev_ops */
+void allmulticast_enable(struct ix_rte_eth_dev *dev);
+void dev_infos_get(struct ix_rte_eth_dev *dev, struct ix_rte_eth_dev_info *dev_info);
+int link_update(struct ix_rte_eth_dev *dev, int wait_to_complete);
+void promiscuous_disable(struct ix_rte_eth_dev *dev);
+int fdir_add_perfect_filter(struct ix_rte_eth_dev *dev, struct rte_fdir_filter *fdir_ftr, uint16_t soft_id, uint8_t rx_queue, uint8_t drop);
+int fdir_remove_perfect_filter(struct ix_rte_eth_dev *dev, struct rte_fdir_filter *fdir_ftr, uint16_t soft_id);
+int rss_hash_conf_get(struct ix_rte_eth_dev *dev, struct ix_rte_eth_rss_conf *ix_reta_conf);
+void mac_addr_add(struct ix_rte_eth_dev *dev, struct eth_addr *mac_addr, uint32_t index, uint32_t vmdq);
+
+/* driver-independent helper functions */
+struct rte_eth_fdir_filter;
```

Comment:
How should we solve that DPDK dependency for the common helper function? Options that I see are:

1) just forward-declare the struct like currently
2) copy-paste this struct + other needed structs in `ix/ethdev.h`
3) include dpdk `ethdev.h` but then we need to put the `undef` for all ix redefinitions like in `.c` files
4) don't extract that helper function here and duplicate it in the drivers